hw/snitch_cluster/src/snitch_cluster_wrapper.sv.tpl

```diff
@@ -392,6 +393,39 @@ for i in range(len(cfg['cores'])):
       total_snax_narrow_ports += snax_narrow_tcdm_ports
       total_snax_wide_ports += snax_wide_tcdm_ports
 
+  elif ('snax_xdma_cfg' in cfg['cores'][i]):
+    snax_xdma_flag = True
+    xdma_cfg = cfg['cores'][i]['snax_xdma_cfg']
+    # Note that the order is from last core to the first core
+    snax_narrow_tcdm_ports_list.append(round(cfg['dma_data_width'] / cfg['data_width']) << 1)
+    snax_wide_tcdm_ports_list.append(0)
+
+    # Prepare accelerator tags
+    xdma_instance_name = "xdma"
+    for key, value in xdma_cfg.items():
+      if key.startswith('has_'):
+        xdma_instance_name += ("_" + key[4:])
+    curr_snax_acc = ''
+    curr_snax_acc = "i_snax_core_" + str(i) + "_" + xdma_instance_name
+
+    # Set tcdm offset ports
+    snax_narrow_tcdm_ports = round(cfg['dma_data_width'] / cfg['data_width']) << 1
+    snax_wide_tcdm_ports = 0
+    snax_tcdm_ports = snax_narrow_tcdm_ports + snax_wide_tcdm_ports
+    tcdm_offset_stop += snax_tcdm_ports
+
+    # Save settings in the dictionary
+    snax_acc_dict[curr_snax_acc] = {
+          'snax_acc_name': xdma_instance_name,
+          'snax_tcdm_ports': snax_tcdm_ports,
+          'snax_tcdm_offset_start': tcdm_offset_start,
+          'snax_tcdm_offset_stop': tcdm_offset_stop
+        }
+    tcdm_offset_start += snax_tcdm_ports
+    prefix_snax_count += 1
+    total_snax_narrow_ports += snax_narrow_tcdm_ports
+    total_snax_wide_ports += snax_wide_tcdm_ports
+
```

Comment:
I find this a bit odd, so when there is an accelerator, we won't use the `snax_xdma` ? I thought we use accelerators AND the xDMA too right? 

Was the intent of the SNAX xDMA to be a stand-alone accelerator? Because that's what the logic in this part says. The first condition `if ('snax_acc_cfg' in cfg['cores'][i]):` holds true then the SNAX xDMA won't be generated.

Reply:
Line 425 removed.